src/main/java/pro/belbix/ethparser/repositories/HarvestRepository.java

```diff
@@ -196,4 +196,29 @@ HarvestDTO findFirstByVaultAndBlockDateLessThanEqualAndIdNotOrderByBlockDateDesc
         + "order by t.blockDate")
     List<HarvestDTO> fetchAllByPeriod(@Param("from") long from, @Param("to") long to);
 
+    List<HarvestDTO> findAllByMethodNameOrderByBlockDate(String methodName);
+
+    List<HarvestDTO> findAllByMethodNameAndBlockDateGreaterThanOrderByBlockDate(String methodName, long blockDate);
+
+    @Query("select t from HarvestDTO t where "
+        + "t.owner = :owner "
+        + "and t.vault = :vault "
+        + "and t.blockDate <= :blockDate "
+        + "and t.blockDate > :lastFullWithdrawBlockDate " 
+        + "order by t.blockDate")
+    List<HarvestDTO> fetchPeriodForOwnerAndVault(@Param("owner") String owner,
```

Comment:
this method can be reusable
better naming is fetchByOwnerAndVaultAndPeriod(@Param("from") long from, @Param("to") long to);